### PR TITLE
pkg/nimble: prepare for building without DEVELHELP

### PR DESCRIPTION
**Pull request by haukepetersen**
### Contribution description
As stated found in #10181, `examples/nimble_gatt` does not build with `DEVELHELP=0` currently. This PR is one step in fixing that. This PR does however not go all the way, as there is also code in nimble that needs to be patched (see https://github.com/apache/mynewt-nimble/pull/225).

So this is my proposed way to get there:
- we merge this PR (as it does not break anything that worked before)
- merge the fix to the riot npl layer in nimble
- make RIOT use the latest master of nimble once the fix has been merged

### Testing procedure
In short: verify that the example is still building as before (with `DEVELHELP=1`), so this PR does not break anything that worked before.
In depth: one would need to switch the nimble package to the branch referenced below and verify a succesfull build with `DEVELHELP=0`

### Issues/PRs references
#10181
https://github.com/apache/mynewt-nimble/pull/225

### CODE REVIEW

**Comment by haukepetersen** [comment]
update:
- apache/mynewt-nimble#225 was merged.
- version bump for the nimble pkt in #10203

**Comment by haukepetersen** [comment]
rebased and adapted the disabled compiler warnings for nimble a bit more, now it should also build fine with `clang` for `DEVELHELP=0`.

Still rebased on (so waiting on) #10203

**Comment by haukepetersen** [comment]
rebased

**Comment by haukepetersen** [comment]
all dependencies are merged.

**Comment by haukepetersen** [comment]
all green -> go
